### ISSUE TITLE
loss=nan when resuming fine-tuning with model_updated.pt

### ISSUE THREAD

**Issue by ghost** [pfnet-research/GenerRNA#7]
I am trying to fine-tune the model_updated.pt checkpoint following the procedure in the HuggingFace Readme.
Even after ensuring my input data is correct and does not contain unusual tokens, the training still fails. When I run train.py with the recommended resume settings, the process starts at iter 178000, but the loss is immediately nan on the very first step.

Here is the log output:
```
gradient_accumulation_steps: 16, batch_size: 16, 
block_size: 256, 
n_layer: 24, n_head: 16, n_embd: 1024, dropout: 0.1, bias: False, 
learning_rate: 0.0001, max_iters: 1000000, 
weight_decay: 0.1, beta1: 0.9, beta2: 0.95, grad_clip: 1.0, decay_lr: False, 
warmup_iters: 2000, lr_decay_iters: 1000000, 
min_lr: 0.0001, backend: nccl, device: cuda,
 dtype: float16, compile: True
meta_vocab_size: 1024
training data: tokenized_psi/
ddp_world_size: 1
tokens per iteration will be: 65,536
Resuming training from psi_ft/
number of parameters: 303.09M
/data/samanthafs/scratch/lab_mguell/mdivincenzo/Documents/T-Bio/psi-library/GenerRNA/train.py:148: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler(enabled=(dtype == 'float16' or dtype == 'bfloat16'))
num decayed parameter tensors: 98, with 303,300,608 parameters
num non-decayed parameter tensors: 49, with 50,176 parameters
using fused AdamW: False
compiling the model... (takes a ~minute)
iter 178000: loss nan, time 42898.31ms, lr 0.0001, mfu -100.00%
iter 178001: loss nan, time 1752.09ms, lr 0.0001, mfu -100.00%
iter 178002: loss nan, time 1752.09ms, lr 0.0001, mfu -100.00%
iter 178003: loss nan, time 1752.39ms, lr 0.0001, mfu -100.00%
iter 178004: loss nan, time 1752.25ms, lr 0.0001, mfu -100.00%
iter 178005: loss nan, time 1752.28ms, lr 0.0001, mfu 22.70%
```
